Content.Server/Shuttles/HelmsmanSystem.cs

```diff
@@ -0,0 +1,156 @@
+using Content.Server.Alert;
+using Content.Server.Power.Components;
+using Content.Shared.ActionBlocker;
+using Content.Shared.Alert;
+using Content.Shared.Interaction;
+using Content.Shared.Notification.Managers;
+using Content.Shared.Shuttles;
+using Robust.Shared.GameObjects;
+
+namespace Content.Server.Shuttles
+{
+    internal sealed class HelmsmanSystem : SharedHelmsmanSystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<HelmsmanConsoleComponent, ComponentShutdown>(HandleHelmsmanShutdown);
+            SubscribeLocalEvent<PilotComponent, ComponentShutdown>(HandlePilotShutdown);
+            SubscribeLocalEvent<HelmsmanConsoleComponent, ActivateInWorldEvent>(HandleHelmsmanInteract);
+            SubscribeLocalEvent<PilotComponent, MoveEvent>(HandlePilotMove);
```

Comment:
i dont like that this is called every time any human moves (assuming they would all have a pilotcomponent), you could maybe add a comp to the entity that's currently piloting a helmsman console and then subscribe directed to that instead? (and remove the comp when they move away, etc)

Reply:
if you do that probably just move all the PilotComponent data onto that and then use a different metric for whether an entity can pilot the shuttle (caninteract?)

Reply:
Problem is that RemoveComponent is currently bugged and doesn't sync the removal from server to client so I hadn't coded around this yet.